src/security_context.rs

```diff
@@ -71,7 +72,7 @@ impl SecurityContext {
                     Ok((SecurityContext(ctxt), Some(ContextBuffer(outbuf[0]))))
                 }
                 err => {
-                    Err(io::Error::from_raw_os_error(err as i32))
+                    Err((io::Error::from_raw_os_error(err as i32), SecurityContext(ctxt)))
```

Comment:
The context object isn't valid if `InitializeSecurityContextW` fails, so we shouldn't be returning it here.